### ISSUE TITLE
False positive for npm module "fresh"

### ISSUE THREAD

**Issue by simonknittel** [anchore/grype#295]
**What happened**: The npm module called "[fresh](https://www.npmjs.com/package/fresh)" gets flagged with [CVE-2013-1779](https://www.whitesourcesoftware.com/vulnerability-database/CVE-2013-1779) which describes an CVE for a theme in Drupal, completely unrelated.

**What you expected to happen**: No CVE warning for fresh.

**How to reproduce it (as minimally and precisely as possible)**: Create a package.json with fresh as dependency and installed in node_modules. Then run grype.

**Anything else we need to know?**: Fresh is downloaded 17m times per week according to npm.

**Environment**:
- Output of `grype version`: 0.9.0
- OS (e.g: `cat /etc/os-release` or similar): ubuntu-20.04 via GitHub Actions

**Comment by luhring** [comment]
@simonknittel Good catch, thanks for reporting this! This looks like a case where our generated CPEs are matching NVD data too broadly — in this case via `"cpe:2.3:a:*:fresh:0.5.2:*:*:*:*:node.js:*:*"`

For more context...

Syft's generated CPEs:
`"cpe:2.3:a:fresh:fresh:0.5.2:*:*:*:*:node.js:*:*"`
`"cpe:2.3:a:fresh:fresh:0.5.2:*:*:*:*:nodejs:*:*"`
`"cpe:2.3:a:*:fresh:0.5.2:*:*:*:*:node.js:*:*"`
`"cpe:2.3:a:*:fresh:0.5.2:*:*:*:*:nodejs:*:*"`
`"cpe:2.3:a:fresh:fresh:0.5.2:*:*:*:*:*:*:*"`
`"cpe:2.3:a:*:fresh:0.5.2:*:*:*:*:*:*:*"`

NVD link: https://nvd.nist.gov/vuln/detail/CVE-2013-1779

**Comment by goranoberg** [comment]
In what I think is another example of the same bug, the npm package ``connect`` seems to be mistaken for Adobe Connect and get flagged for a number of vulnerabilities, e.g. the Critical [CVE-216-0949](https://nvd.nist.gov/vuln/detail/CVE-2016-0949).

Steps to reproduce:
```
mkdir test
cd test
rm -fr *
npm install connect@3.7.0
grype dir:.
syft -q packages -o json | grep -A1 -B1 ':connect:'
```


grype give 33 matching CVEs of which I looked closer at the 5 Critical ones. All those 5 was false positives, instead describing Adobe Connect.

syft will identify the installed component as
```
   "cpes": [
    "cpe:2.3:a:connect:connect:3.7.0:*:*:*:*:*:*:*",
    "cpe:2.3:a:*:connect:3.7.0:*:*:*:*:*:*:*"
   ],
```

In CVE-2016-0949 known affected software are listed as ``cpe:2.3:a:adobe:connect:*:*:*:*:*:*:*:*``. Without having looked at the code, I suspect there is a chance that it's shorter of the two ids from syft that leads to these false positives

----

Version information:
```
Application:          grype
Version:              0.19.0
BuildDate:            2021-09-15T14:17:43Z
GitCommit:            7c082c2162fa36f9fd0cd2b34dfcb86348036275
GitTreeState:         clean
Platform:             darwin/amd64
GoVersion:            go1.16.8
Compiler:             gc
Supported DB Schema:  3

Application:   syft
Version:       0.23.0
BuildDate:     2021-09-13T17:48:43Z
GitCommit:     ddfc8e20c0f370d25e7d032b23bef2e3d519506c
GitTreeState:  clean
Platform:      darwin/amd64
GoVersion:     go1.16.7
Compiler:      gc
```

**Comment by luhring** [comment]
@wagoodman Curious for your thoughts when you get a chance. Something interesting I found is that with Syft v0.21.0 produces CPEs that **_don't_** yield this false positive, but this is no longer the case as of Syft v0.23.0.

Syft v0.21.0:
```
"cpes": [
  "cpe:2.3:a:connect:connect:3.7.0:*:*:*:*:node.js:*:*",
  "cpe:2.3:a:connect:connect:3.7.0:*:*:*:*:nodejs:*:*",
  "cpe:2.3:a:connect:connect:3.7.0:*:*:*:*:*:*:*"
],
```

Syft v0.23.0:

```
"cpes": [
  "cpe:2.3:a:connect:connect:3.7.0:*:*:*:*:*:*:*",
  "cpe:2.3:a:*:connect:3.7.0:*:*:*:*:*:*:*"
],
```

So in other words, the FPs return in Syft v0.23.0.

**Comment by luhring** [comment]
@wagoodman If I understand the updated CPE generation approach, we can control on a per-ecosystem basis whether we use `*` in the vendor field? I'm wondering whether it's net-better to use or not use `*` for NPM packages... (using the above as one data point)

**Comment by wagoodman** [comment]
> we can control on a per-ecosystem basis whether we use * in the vendor field?

correct, and we should consider re-evaluating using `*` as a vendor candidate on a pseudo-continual basis. It seems that relative to the vendors we're seeing in CPEs in the NVD data there isn't always a way to find a match due to there not being sufficient data in the package metadata (For Java, this isn't a problem, as we usually just need to add another spot to consider, but at least the data is there... for some other ecosystems it has been a problem).

>  I'm wondering whether it's net-better to use or not use * for NPM packages

I wonder the same thing! I think we need to take a look at a wide set of images with node packages and do a more detailed look to characterize the behavior today and to start investigating if there are better options.

**Comment by willmurphyscode** [comment]
Added `changelog-ignore` because this was fixed in `0.60.0` and so shouldn't be included in the current release's release notes.
